docs/guides/typescript.mdx

```diff
@@ -31,13 +31,45 @@ const numAtom = atom<number | null>(0)
 const arrAtom = atom<string[]>([])
 ```
 
-### Derived atoms are also type inferred and explicitly typed
+### Derived atoms can mostly have their types inferred
 
 ```ts
 const asyncStrAtom = atom(async () => 'foo')
 const writeOnlyAtom = atom(null, (_get, set, str: string) => set(fooAtom, str))
-const readWriteAtom = atom<string, number>(
+const readWriteAtom = atom(
   (get) => get(strAtom),
+  (_get, set, num: number) => set(strAtom, String(num))
+)
+```
+
+### Derived atoms can also be explicitly typed
+
+If you encounter a situation where you need or want to explicitly
+type your derived atoms, you can do that as well.
```

Comment:
I wonder if we can emphasize this more.
We should probably show more example with type inference.
(Currently, the section size seems bigger for manual typing.)